### PR TITLE
[issue 6] LRU cache instead of unlimited object cache

### PR DESCRIPTION
**Pull request by nmccready**
Attempt at resolving #6 
- LRU cache instead of unlimited object cache
- removed array monads to use native for loop for more speed

### CODE REVIEW

File: coffeelint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "max_line_length": {
+    "value": 94,
```
Comment (nmccready) [inline coffeelint.json:3]: Chose a value that allowed `lib/linter-coffee-variables.coffee` to have no errors.

File: lib/linter-coffee-variables.coffee [hunk 2]
```diff
@@ -1,9 +1,12 @@
 debug = require './debug'
+LRU = require('lru')
+coffee = require 'coffee-script'
 
-cache = {}
+cache = new LRU
```
Comment (nmccready) [inline lib/linter-coffee-variables.coffee:5]: LRU as discussed in issue 6.

File: lib/linter-coffee-variables.coffee [hunk 3]
```diff
@@ -103,12 +110,14 @@ _addOriginalCodePosition = (sourceMap, variables) -> (error) ->
...
 
 _lookUpVariablePosition = (variableName, variables) ->
-  variables
-    .filter (v) ->
-      v.name is variableName
-    .map (v) ->
-      line   : v.line
-      column : v.column
+  mappedVariables = []
+  for v in variables
```
Comment (nmccready) [inline lib/linter-coffee-variables.coffee:114]: Yes it is not sexy, but basic `for loop` will be faster for large files.

Comment (nmccready) [inline lib/linter-coffee-variables.coffee:114]: Just run this in node and see for yourself.
http://codereview.stackexchange.com/questions/49549/performance-for-foreach-and-for-key

**Comment by nmccready** [comment]
@markwainwright please review

**Review by markwainwright — APPROVED** [review]
Hi @nmccready - sorry for leaving this for so long. I don't really code in CoffeeScript any more – but that's no excuse!

Changes look good. Thanks so much for taking the time to make them.

I'll merge this in and publish a release shortly.

File: lib/linter-coffee-variables.coffee [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
+LRU = require('lru')
+coffee = require 'coffee-script'
 
-cache = {}
+cache = new LRU
+  max: 10
+  maxAge: 60 * 60 * 1000
 
 _compileToJS = (coffeeSource) ->
-  coffee = require 'coffee-script'
```
Comment (markwainwright) [inline lib/linter-coffee-variables.coffee:6]: As weird as this is, I originally did it to attempt to cut down the time the plugin took to initialize, which slowed down Atom's initial load. But I'm willing to put it back to the top of the file and see if it really makes a difference.

**Pull request merged**